File: 4_Quarta/4_Exo_Quarta_eg.mjs
import namelist from './4_Exo_Quarta.json' assert {type: 'json'};

function groupes(k) {
    if (k < 1) { throw new Error("nope"); }
    let l = [...namelist];
    l.sort((a, b) => Math.random() - 0.5);
    let result = [];
    while (l.length > 0) {
        result.push(l.splice(0, k));
    }
    return result;
}

const k = 4;
console.log(groupes(k));

File: 4_Quarta/4_Exo_Quarta_eg.py
import json
import random

with open('4_Exo_Quarta.json') as j:
        l = json.load(j)

def groupe(k):
    if k < 1: raise ValueError("Nope")
    data = l.copy()
    random.shuffle(data)
    result = []
    while len(data) > 0:
        g = []
        for i in range(k):
            g.append(data.pop())
            if len(data) == 0: break
        result.append(g)
    return result

k = int(input("define value of k: "))
print(groupe(k))